src/java.base/windows/classes/java/io/WinNTFileSystem.java

```diff
@@ -365,24 +365,26 @@ public boolean isInvalid(File f) {
         if (f.getPath().indexOf('\u0000') >= 0)
             return true;
 
+        // Invalid if the pathname has a trailing space
+        String pathname = f.getPath();
+        int len = pathname.length();
+        if (len > 0 && pathname.charAt(len - 1) == ' ')
+            return true;
```

Comment:
Can you think about "foo \bar" case too?

Reply:
I am not sure what you mean here as file paths _can_ contain (s) space(s).

Reply:
The question is about directories in the path that end with space. So yes, the file path may contain spaces but the proposed check is only looking at the end last element.

Reply:
From experimentation with C++ code is VS, it appears that `CreateDirectoryW` will create a directory with either a leading or trailing space in its name, whereas `createFileW` will create a file with a leading space in its name, but a trailing space is trimmed.

Reply:
> From experimentation with C++ code is VS, it appears that `CreateDirectoryW` will create a directory with either a leading or trailing space in its name, whereas `createFileW` will create a file with a leading space in its name, but a trailing space is trimmed.

Right, the Windows APIs are very inconsistent on this, and why the Windows implementation of Path will always reject input where an element has a trailing space.

Reply:
It turns out that a case such as `foo \bar\gus` already fails in `File.mkdirs` due to directory creation itself, so an explicit check for a trailing space in directory elements does not appear to be necessary, although having the failure occur in `isInvalid` would be more fail-fast.

Reply:
The benefit of doing it before attempting any I/O operation is avoids the unpredictable behavior of the Win32 calls. Otherwise, I think it means through each Win32 call used by WinNTFileSystem and seeing how it behaves.